tests/test_query.py

```diff
@@ -142,3 +142,65 @@ def test_elasticsearch_query_min_score(app):
     search_dict = q.to_dict()
     assert 'min_score' in search_dict
     assert search_dict['min_score'] == app.config['SEARCH_RESULTS_MIN_SCORE']
+
+
+def _test_original_index_is_stored_when_prefixing(q, prefix_value, index):
+    """Test original index is stored."""
+
+    q = q.query(Q('match', title='Higgs'))
+    q = q.sort()
+    search_dict = q.to_dict()
+    assert 'query' in search_dict
+
+    assert q._index[0] == '{}{}'.format(prefix_value, index)
+    assert q._original_index[0] == index
+
+
+def test_prefix_index_from_meta(app):
+    """Test that index is prefixed when you use `Meta.index` field."""
+    class TestSearch(RecordsSearch):
+        class Meta:
+            index = 'myindex'
+
+    prefix_value = 'myprefix-'
+    index_value = TestSearch.Meta.index
+    app.config['SEARCH_INDEX_PREFIX'] = prefix_value
+
+    q = TestSearch()
+    _test_original_index_is_stored_when_prefixing(q, prefix_value, index_value)
+
+
+def test_prefix_index_from_kwargs(app):
+    """Test that index is prefixed when pass it through kwargs."""
+    prefix_value = 'myprefix-'
+    index_value = 'myindex'
+    app.config['SEARCH_INDEX_PREFIX'] = prefix_value
+
+    q = RecordsSearch(index=index_value)
+    _test_original_index_is_stored_when_prefixing(q, prefix_value, index_value)
+
+
+def test_prefix_index_list(app):
+    """Test that index is prefixed when pass it through kwargs."""
+    prefix_value = 'myprefix-'
+    index_value = ['myindex', 'myanotherindex']
+    app.config['SEARCH_INDEX_PREFIX'] = prefix_value
+
+    q = RecordsSearch(index=index_value)
+    q.query()
+    assert q._index == ['{}{}'.format(prefix_value,
+                                      _index) for _index in index_value]
+    assert q._original_index == index_value
+
+
+def test_prefix_multi_index_string(app):
+    """Test that index is prefixed when pass it through kwargs."""
+    prefix_value = 'myprefix-'
+    index_value = 'myindex,myanotherindex'
+    app.config['SEARCH_INDEX_PREFIX'] = prefix_value
+
+    q = RecordsSearch(index=index_value)
+    q.query()
```

Comment:
I think you have to pass some query here and then save the result in `q` to properly test the prefixing